### PR TITLE
Add test for selection and_then

### PR DESCRIPTION
**Pull request by Ted-Jiang**
# Which issue does this PR close?

Related #2925.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: parquet/src/arrow/arrow_reader/selection.rs [hunk 1]
```diff
@@ -455,11 +455,27 @@ mod tests {
...
                 RowSelector::skip(1),
                 RowSelector::select(1),
-                RowSelector::skip(4)
+                RowSelector::skip(4),
             ]
         );
     }
 
+    #[test]
+    fn test_and_2() {
```
Comment (Ted-Jiang) [inline parquet/src/arrow/arrow_reader/selection.rs:464]: seems this is a bug 🤔 @tustvold

Comment (tustvold) [reply parquet/src/arrow/arrow_reader/selection.rs:464]: Yup, that doesn't look correct :sweat_smile:

Comment (tustvold) [reply parquet/src/arrow/arrow_reader/selection.rs:464]: The test output should be

```
RowSelector::select(3),
RowSelector::skip(33),
RowSelector::select(3),
RowSelector::skip(33),
```

Although it is worth noting that the b selection is a little bit odd as it covers a range larger than that selected by a?

**Comment by tustvold** [comment]
Created https://github.com/apache/arrow-rs/pull/2928/files to improve the panic messages

File: parquet/src/arrow/arrow_reader/selection.rs (hunk 1 above)
Comment (Ted-Jiang) [reply parquet/src/arrow/arrow_reader/selection.rs:464]: @tustvold  
> Although it is worth noting that the b selection is a little bit odd as it covers a range larger than that selected by a? b is a selection of the 6 rows already selected by a

Is this assumption:  last one must select less row than  before one, only for `filter_pushdown`,
I create this two selections by apply filter on pageIndex.

Comment (tustvold) [reply parquet/src/arrow/arrow_reader/selection.rs:464]: Yes, as per the doc comment, and_then is for computing the resulting RowSelection when the provided RowSelection is the result of the computation "under" another. We can, and probably should add a separate `and` for combining together row selections

Comment (Ted-Jiang) [reply parquet/src/arrow/arrow_reader/selection.rs:464]: >as per the doc comment
i missunder the computed under `self` 😂

>  add a separate and for combining together row selections
Agree! will add this.

**Pull request closed without merging**